rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1635,119 +1659,133 @@ private Optional<Federation> getFederationFromPublicKey(BtcECKey federatorPublic
+    }
 
-        // All signatures are correct. Proceed to signing
-        for (int i = 0; i < numInputs; i++) {
-            Sha256Hash sighash = sighashes.get(i);
+    private boolean sign(
+        BtcECKey federatorBtcPublicKey,
+        List<TransactionSignature> txSigs,
+        List<Sha256Hash> sigHashes,
+        Keccak256 rskTxHash,
+        BtcTransaction btcTx) {
+
+        boolean signed = false;
+        for (int i = 0; i < sigHashes.size(); i++) {
+            Sha256Hash sighash = sigHashes.get(i);
             TransactionInput input = btcTx.getInput(i);
             Script inputScript = input.getScriptSig();
 
-            boolean alreadySignedByThisFederator = BridgeUtils.isInputSignedByThisFederator(
-                    federatorBtcPublicKey,
-                    sighash,
-                    input);
+            boolean alreadySignedByThisFederator =
+                BridgeUtils.isInputSignedByThisFederator(federatorBtcPublicKey, sighash, input);
 
             // Sign the input if it wasn't already
-            if (!alreadySignedByThisFederator) {
-                try {
-                    int sigIndex = inputScript.getSigInsertionIndex(sighash, federatorBtcPublicKey);
-                    inputScript = ScriptBuilder.updateScriptWithSignature(inputScript, txSigs.get(i).encodeToBitcoin(), sigIndex, 1, 1);
-                    input.setScriptSig(inputScript);
-                    logger.debug("Tx input {} for tx {} signed.", i, new Keccak256(rskTxHash));
-                    signed = true;
-                } catch (IllegalStateException e) {
-                    Federation retiringFederation = getRetiringFederation();
-                    if (getActiveFederation().hasBtcPublicKey(federatorBtcPublicKey)) {
-                        logger.debug("A member of the active federation is trying to sign a tx of the retiring one");
-                        return;
-                    } else if (retiringFederation != null && retiringFederation.hasBtcPublicKey(federatorBtcPublicKey)) {
-                        logger.debug("A member of the retiring federation is trying to sign a tx of the active one");
-                        return;
-                    }
-                    throw e;
-                }
-            } else {
-                logger.warn("Input {} of tx {} already signed by this federator.", i, new Keccak256(rskTxHash));
+            if (alreadySignedByThisFederator) {
+                logger.warn("Input {} of tx {} already signed by this federator.", i, rskTxHash);
                 break;
             }
-        }
 
-        if(signed && activations.isActive(ConsensusRule.RSKIP326)) {
-            eventLogger.logAddSignature(federatorMember, btcTx, rskTxHash);
+            try {
+                int sigIndex = inputScript.getSigInsertionIndex(sighash, federatorBtcPublicKey);
+                inputScript = ScriptBuilder.updateScriptWithSignature(inputScript, txSigs.get(i).encodeToBitcoin(), sigIndex, 1, 1);
```

Comment:
I'd consider using `getScriptSigWithSignature` method. Internally it calls `updateScriptWithSignature` but receives less params so no need to pass those `1, 1` value that no one knows what they are.

Double check just in case that the behavior is the same

Reply:
updated! Thanks for noticing it